squidb-processor/src/com/yahoo/squidb/processor/plugins/PluginManager.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the Apache 2.0 License.
+ * See the accompanying LICENSE file for terms.
+ */
+package com.yahoo.squidb.processor.plugins;
+
+import com.yahoo.aptutils.utils.AptUtils;
+import com.yahoo.squidb.processor.data.ModelSpec;
+import com.yahoo.squidb.processor.plugins.defaults.ConstantCopyingPlugin;
+import com.yahoo.squidb.processor.plugins.defaults.ConstructorPlugin;
+import com.yahoo.squidb.processor.plugins.defaults.ImplementsPlugin;
+import com.yahoo.squidb.processor.plugins.defaults.ModelMethodPlugin;
+import com.yahoo.squidb.processor.plugins.defaults.properties.InheritedModelSpecFieldPlugin;
+import com.yahoo.squidb.processor.plugins.defaults.properties.TableModelSpecFieldPlugin;
+import com.yahoo.squidb.processor.plugins.defaults.properties.ViewModelSpecFieldPlugin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.tools.Diagnostic;
+
+/**
+ * This class maintains a list of known/enabled {@link Plugin} classes. Plugins available by default include
+ * {@link ConstructorPlugin} for generating model constructors, {@link ImplementsPlugin} for allowing models to
+ * implement interfaces, {@link ModelMethodPlugin} for copying methods from the model spec to the model, and the three
+ * property generator plugins ({@link TableModelSpecFieldPlugin}, {@link ViewModelSpecFieldPlugin}, and
+ * {@link InheritedModelSpecFieldPlugin}).
+ * <p>
+ * This class also manages option flags for plugins. The default plugins for constructors, interfaces, and model
+ * methods can all be disabled using a flag. Other flags allow disabling default content values, disabling the
+ * convenience getters and setters that accompany each property, and preferring user-defined plugins to the default
+ * plugins. Options are passed as a bitmask to the processor using the key "squidbOptions".
+ */
+public class PluginManager {
+
+    private final AptUtils utils;
+    private final int optionFlags;
+    private List<Class<? extends Plugin>> highPriorityPlugins = new ArrayList<Class<? extends Plugin>>();
+    private List<Class<? extends Plugin>> normalPriorityPlugins = new ArrayList<Class<? extends Plugin>>();
+    private List<Class<? extends Plugin>> lowPriorityPlugins = new ArrayList<Class<? extends Plugin>>();
+
+    public enum PluginPriority {
+        LOW,
+        NORMAL,
+        HIGH
+    }
+
+    /**
+     * Flag for disabling the default constructors generated in each model class
+     */
+    public static final int OPTIONS_DISABLE_DEFAULT_CONSTRUCTORS = 1;
```

Comment:
Bit flags might be hard since the constants can't be referenced in build scripts. Maybe comma-separated string constants? Those are easier to copy-paste.